### ISSUE TITLE
Removed Songs from Playlist - Did not get re-added

### ISSUE THREAD

**Issue by ace101boss** [megabulk/Fix-for-iTunes-file-truncation#1]
Ran the script on OS Mojave (10.14) with iTunes v12.9.0.164 and all the songs from my playlist got removed and did not get re-added. Running the script on the library song list could not get past removing and re-adding the first song.

**Comment by megabulk** [comment]
Thanks for the feedback. I haven’t noticed the problem with truncated songs for a while, so I haven’t bothered to test the script lately. It worked for me when I needed it.
Sorry it didn’t work for you, and I hope it wasn’t too inconveniencing.

**Comment by ace101boss** [comment]
> Thanks for the feedback. I haven’t noticed the problem with truncated songs for a while, so I haven’t bothered to test the script lately. It worked for me when I needed it. Sorry it didn’t work for you, and I hope it wasn’t too inconveniencing.
> […](#)

Thanks for the reply on this, I got around the issue by re-importing the playlists from my iPhone so not too much an issue. For the other issue where running the script on the library song list and it removing and adding the first song in an infinite loop - I got around this by selecting all songs (CMD + A) and then running the script. Of course this action also removed all the songs from my playlists again. I guess for anybody that comes across your helpful script going forward - back up/export the playlists before running the script.

Your script helped to resolve my issues of the songs cutting off early so I appreciate you building it in the first place!
